pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/BrokersBase.java

```diff
@@ -343,41 +350,81 @@ public void healthCheck(@Suspended AsyncResponse asyncResponse,
                         return client.newProducer(Schema.STRING).topic(topicName).createAsync()
-                                        .thenCombine(client.newReader(Schema.STRING).topic(topicName)
+                                .thenCombine(client.newReader(Schema.STRING).topic(topicName)
                                         .startMessageId(MessageId.latest).createAsync(), (producer, reader) ->
-                                                        producer.sendAsync(messageStr).thenCompose(__ ->
-                                                                healthCheckRecursiveReadNext(reader, messageStr))
-                                                        .thenCompose(__ -> {
-                                                            List<CompletableFuture<Void>> closeFutures =
-                                                                    new ArrayList<>();
-                                                            closeFutures.add(producer.closeAsync());
-                                                            closeFutures.add(reader.closeAsync());
-                                                            return FutureUtil.waitForAll(closeFutures);
-                                                        })
-                                        ).thenAccept(ignore -> {});
+                                        producer.sendAsync(messageStr).thenCompose(__ ->
+                                                        healthCheckRecursiveReadNext(reader, messageStr))
+                                                .thenCompose(__ -> {
+                                                    List<CompletableFuture<Void>> closeFutures =
+                                                            new ArrayList<>();
+                                                    closeFutures.add(producer.closeAsync());
+                                                    closeFutures.add(reader.closeAsync());
+                                                    return FutureUtil.waitForAll(closeFutures);
+                                                })
+                                ).thenAccept(ignore -> {});
                     } catch (PulsarServerException e) {
                         LOG.error("[{}] Fail to run health check while get client.", clientAppId());
                         throw new RestException(e);
                     }
                 });
     }
 
+    private CompletableFuture<Void> tryCleanPreviousSubscriptions(Topic topic) {
+        // clean all subscriptions
+        return FutureUtil.waitForAll(topic.getSubscriptions().values()
+                .stream().filter(subscription ->
+                // All system topics are using compaction, even though is not explicitly set in the policies.
+                        !subscription.getName().equals(Compactor.COMPACTION_SUBSCRIPTION))
+                .map(Subscription::delete).collect(Collectors.toList()))
```

Comment:
Why add filter ?

Reply:
@Technoboy- 
Because all the system topic has compaction subscription that name is ``__compaction``, every time we call health check, we will create a new reader with a subscription name's ``reader-xxx``. we just need to remove the reader subscriptions and don't need to remove ``__compaction`` subscription.
I'm not sure if I explain clearly, please let me know what you think, very thanks~

Reply:
Ok, we can add this.